libs/azure-api-client/src/GuzzleClientFactory.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\AzureApiClient;
+
+use GuzzleHttp\Client as GuzzleClient;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\MessageFormatter;
+use GuzzleHttp\Middleware;
+use Keboola\AzureApiClient\Exception\ClientException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Range;
+use Symfony\Component\Validator\Constraints\Url;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\Validation;
+
+class GuzzleClientFactory
+{
+    private const DEFAULT_USER_AGENT = 'Azure PHP Client';
+    private const DEFAULT_BACKOFF_RETRIES = 10;
+    private const AZURE_THROTTLING_CODE = 429;
+    private const ALLOWED_OPTIONS = ['backoffMaxTries', 'userAgent', 'handler', 'logger'];
+
+    private LoggerInterface $logger;
+
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    public function getLogger(): LoggerInterface
+    {
+        return $this->logger;
+    }
+
+    public function getClient(string $baseUrl, array $options = []): GuzzleClient
```

Comment:
array shape? zjednodušuje to tu validaci https://github.com/keboola/billing-api-php-client/blob/main/src/InternalClient.php

Reply:
jop, je pridany v ramci nasledujiciho PR https://github.com/keboola/platform-libraries/pull/43/commits/e74abe4f749b6beb5f2eaea7c371b18185ab65b2#diff-bee3ed1de21607a49610ac8a8c39f94e7446a717f0cf42430748603da283e7e1R40-R47

Reply:
jj